migrations/versions/a528c3c81506_unique_generic_sensor_ids.py

```diff
@@ -0,0 +1,405 @@
+        sa.Column("name", sa.String(80)),
+    )
+    t_markets = sa.Table(
+        "market",
+        sa.MetaData(),
+        sa.Column("id", sa.Integer),
+        sa.Column("name", sa.String(80)),
+    )
+    t_weather_sensors = sa.Table(
+        "weather_sensor",
+        sa.MetaData(),
+        sa.Column("id", sa.Integer),
+        sa.Column("name", sa.String(80)),
+    )
+
+    # Use SQLAlchemy's connection and transaction to go through the data
+    connection = op.get_bind()
+
+    # Get the max id used by assets and markets
+    max_asset_id = get_max_id(connection, "asset")
+    max_market_id = get_max_id(connection, "market")
+    max_weather_sensor_id = get_max_id(connection, "weather_sensor")
+
+    # Select all existing ids that need migrating, while keeping names intact
+    asset_results = connection.execute(
+        sa.select(
+            [
+                t_assets.c.id,
+                t_assets.c.name,
+            ]
+        )
+    ).fetchall()
+    market_results = connection.execute(
+        sa.select(
+            [
+                t_markets.c.id,
+                t_markets.c.name,
+            ]
+        )
+    ).fetchall()
+    weather_sensor_results = connection.execute(
+        sa.select(
+            [
+                t_weather_sensors.c.id,
+                t_weather_sensors.c.name,
+            ]
+        )
+    ).fetchall()
+
+    # Prepare to build a list of new sensors
+    new_sensors = []
+
+    # Iterate over all assets
+    for id_, name in asset_results:
+        # Determine the new id
+        new_id = id_  # assets keep their original ids
+        # Create new Sensors with matching ids
+        new_sensor = Sensor(name=name)
+        new_sensor.id = new_id
+        new_sensors.append(new_sensor)
+
+    # Iterate over all markets
+    for id_, name in market_results:
+        # Determine the new id
+        new_id = id_ + max_asset_id
+        # Update the id
+        connection.execute(
+            t_markets.update().where(t_markets.c.name == name).values(id=new_id)
+        )
+        # Create new Sensors with matching ids
+        new_sensor = Sensor(name=name)
+        new_sensor.id = new_id
+        new_sensors.append(new_sensor)
+
+    # Iterate over all weather sensors
+    for id_, name in weather_sensor_results:
+        # Determine the new id
+        new_id = id_ + max_asset_id + max_market_id
+        # Update the id
+        connection.execute(
+            t_weather_sensors.update()
+            .where(t_weather_sensors.c.name == name)
+            .values(id=new_id)
+        )
+        # Create new Sensors with matching ids
+        new_sensor = Sensor(name=name)
+        new_sensor.id = new_id
+        new_sensors.append(new_sensor)
+
+    # Add the new sensors
+    session = orm.Session(bind=connection)
+    session.add_all(new_sensors)
+    session.commit()
+
+    # Downgrade schema after supporting data upgrade
+    temporary_schema_downgrades()
```

Comment:
The name and comment here are confusing to me. We're doing something specific in that function, changing the behaviour of foreign keys, why such a generic name?